pkg/controller/cloud/node_controller.go

```diff
@@ -412,11 +412,7 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 	if node.Spec.ProviderID == "" {
 		providerID, err := cloudprovider.GetInstanceProviderID(ctx, cnc.cloud, types.NodeName(node.Name))
 		if err == nil {
-			nodeModifiers = append(nodeModifiers, func(n *v1.Node) {
```

Comment:
@ialidzhikov unless I am missing something completely, this change will result in the `providerID` never being set on the node, because the node is updated via a `Get` -> `Modify` -> `Update`.

Reply:
And also please add a oneline godoc on why it has to be set on the `node`, this is super unintuitive and as we can see untested

Reply:
> @ialidzhikov unless I am missing something completely, this change will result in the providerID never being set on the node, because the node is updated via a Get -> Modify -> Update.

Yep, you are right. The providerID won't be set but the Node will be initiliazed. Then I would suggest to keep the nodeModifiers append but also set the .spec.providerID for the `node` variable. WDYT?

Reply:
yes, sgtm

Reply:
> And also please add a oneline godoc on why it has to be set on the node, this is super unintuitive and as we can see untested

@alvaroaleman , I updated the PR, it should be more intuitive now.